internal/castai/castai.go

```diff
@@ -96,7 +102,22 @@ func createHTTPTransport() *http.Transport {
 		IdleConnTimeout:       90 * time.Second,
 		TLSHandshakeTimeout:   5 * time.Second,
 		ExpectContinueTimeout: 1 * time.Second,
+		TLSClientConfig:       tlsConfig,
+	}
+}
+
+func createTLSConfig() (*tls.Config, error) {
+	if cert := config.Get().TLS; cert != nil {
+		certPool := x509.NewCertPool()
+		ok := certPool.AppendCertsFromPEM([]byte(cert.CACertFile))
+		if !ok {
+			return nil, fmt.Errorf("failed to add root certificate to CA pool")
+		}
+		return &tls.Config{
+			RootCAs: certPool,
+		}, nil
 	}
+	return nil, nil
```

Comment:
Style nitpick: the `if` statement can be inverted to reduce indentation. Have the `cert == nil` check as the first check.

Reply:
Reorganised as suggested.